ksql-parser/src/main/java/io/confluent/ksql/parser/tree/StatementProperties.java

```diff
@@ -0,0 +1,311 @@
+    this.key = key;
+
+    this.avroSchemaName = avroSchemaName;
+    this.avroSchemaId = avroSchemaId;
+
+    this.timestampName = timestampName;
+    this.timestampFormat = timestampFormat;
+  }
+
+  public Format getValueFormat() {
+    return valueFormat.value;
+  }
+
+  public String getKafkaTopic() {
+    return kafkaTopic.value;
+  }
+
+  public Optional<String> getKey() {
+    return Optional.ofNullable(key.value);
+  }
+
+  public Optional<SerdeFactory<Windowed<String>>> getWindowType() {
+    return Optional.ofNullable(windowType.value).map(WINDOW_TYPES::get);
+  }
+
+  public Optional<String> getTimestampName() {
+    return Optional.ofNullable(timestampName.value);
+  }
+
+  public Optional<String> getKsqlTopic() {
+    return Optional.ofNullable(ksqlTopic.value);
+  }
+
+  public Optional<Integer> getAvroSchemaId() {
+    return Optional.ofNullable(avroSchemaId.value);
+  }
+
+  public Optional<String> getTimestampFormat() {
+    return Optional.ofNullable(timestampFormat.value);
+  }
+
+  public Optional<String> getAvroSchemaName() {
+    return Optional.ofNullable(avroSchemaName.value);
+  }
+
+  public StatementProperties withSchemaId(final int id) {
+    return new StatementProperties(
+        ksqlTopic,
+        timestampName,
+        windowType,
+        key,
+        kafkaTopic,
+        valueFormat,
+        avroSchemaName,
+        new Property<>(
+            KsqlConstants.AVRO_SCHEMA_ID,
+            new StringLiteral(Integer.toString(id)),
+            ignored -> id),
+        timestampFormat
+    );
+  }
+
+  // CHECKSTYLE_RULES.OFF: CyclomaticComplexity
+  @Override
+  public boolean equals(final Object o) {
+    // CHECKSTYLE_RULES.ON: CyclomaticComplexity
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    final StatementProperties that = (StatementProperties) o;
+    return Objects.equals(kafkaTopic, that.kafkaTopic)
+        && Objects.equals(valueFormat, that.valueFormat)
+        && Objects.equals(key, that.key)
+        && Objects.equals(ksqlTopic, that.ksqlTopic)
+        && Objects.equals(windowType, that.windowType)
+        && Objects.equals(timestampName, that.timestampName)
+        && Objects.equals(timestampFormat, that.timestampFormat)
+        && Objects.equals(avroSchemaName, that.avroSchemaName)
+        && Objects.equals(avroSchemaId, that.avroSchemaId);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects
+        .hash(kafkaTopic, valueFormat, key, ksqlTopic, windowType, timestampName, timestampFormat,
+            avroSchemaName, avroSchemaId);
+  }
+
+  private static final class Property<T> {
+
+    final T value;
+    final String name;
+    final Literal original;
+
+    private Property(final String name, final Literal literal , final Function<Object, T> extract) {
+      this.original = literal;
+      this.value = literal == null ? null : cleanQuotesIfString(extract.apply(literal.getValue()));
```

Comment:
`literal.getValue()` would return a string _without_ the quotes.  So the `cleanQuotesIfString` should not be required.